rest/views.py

```diff
@@ -593,3 +599,145 @@ def perform_destroy(self, instance):
+
+    return {
+        "id": task_id,
+        "status": state,
+        "url": url
+    }
+
+
+class LearningResourceExportTaskList(ListCreateAPIView):
+    """
+    View for export tasks for a user.
+    """
+    serializer_class = LearningResourceExportTaskSerializer
+    permission_classes = (
+        ViewRepoPermission,
+        IsAuthenticated,
+    )
+
+    def get_queryset(self):
+        """Get tasks for this user."""
+        repo_slug = self.kwargs['repo_slug']
+        try:
+            tasks = self.request.session[EXPORT_TASK_KEY][repo_slug]
+        except KeyError:
+            tasks = {}
+
+        return [create_task_result_dict(task) for task in tasks.values()]
+
+    def post(self, request, *args, **kwargs):
+        """
+        Create a new export task for this user.
+
+        Note that this also cancels and clears any old tasks for the user,
+        so there should be only one task in the list at any time.
+
+        The export list will be left alone. Once the list is exported
+        the client may DELETE the export list as a separate REST call.
+        """
+
+        repo_slug = self.kwargs['repo_slug']
+        try:
+            exports = self.request.session[EXPORTS_KEY][repo_slug]
+        except KeyError:
+            exports = []
+
+        # Cancel any old tasks.
+        old_tasks = self.request.session.get(
+            EXPORT_TASK_KEY, {}).get(repo_slug, {})
+        for task_id in old_tasks.keys():
+            AsyncResult(task_id).revoke()
+
+        # Clear task list.
+        if EXPORT_TASK_KEY not in self.request.session:
+            self.request.session[EXPORT_TASK_KEY] = {}
+        self.request.session[EXPORT_TASK_KEY][repo_slug] = {}
+
+        learning_resources = LearningResource.objects.filter(
+            id__in=exports).all()
+        result = export_resources.delay(
+            learning_resources, self.request.user.username)
+
+        if result.successful():
+            url = default_storage.url(result.get())
+        else:
+            url = ""
+
+        # Put new task in session.
+        self.request.session[EXPORT_TASK_KEY][repo_slug][result.id] = {
+            "id": result.id,
+            "initial_state": result.state,
+            "url": url
+        }
+        self.request.session.modified = True
+
+        return Response(
+            {"id": result.id},
+            status=status.HTTP_201_CREATED
+        )
+
+
+class LearningResourceExportTaskDetail(RetrieveAPIView):
+    """
+    Detail view for a user's export tasks.
+    """
+    serializer_class = LearningResourceExportTaskSerializer
+    permission_classes = (
+        ViewRepoPermission,
+        IsAuthenticated,
+    )
+
+    def get_object(self):
+        """
+        Retrieve current information about an export task.
+        """
+        try:
+            task_id = self.kwargs['task_id']
+        except ValueError:
+            raise Http404
+
+        repo_slug = self.kwargs['repo_slug']
```

Comment:
This variable isn't really necessary, since it's only used once, and wouldn't make the line too long.